src/main/kotlin/org/rust/cargo/toolchain/Cargo.kt

```diff
@@ -73,25 +73,24 @@ class Cargo(
         additionalArguments: List<String> = emptyList(),
         environmentVariables: Map<String, String> = emptyMap()
     ): GeneralCommandLine {
-        var args = additionalArguments
         val cmdLine = GeneralCommandLine(pathToCargoExecutable)
             .withWorkDirectory(projectDirectory)
             .withParameters(command)
-            .withEnvironment(CargoConstants.RUSTC_ENV_VAR, pathToRustExecutable)
-            .withEnvironment(environmentVariables)
 
         // Make output colored
         if ((SystemInfo.isLinux || SystemInfo.isMac)
             && command in COLOR_ACCEPTING_COMMANDS
             && additionalArguments.none { it.startsWith("--color") }) {
             cmdLine
-                .withEnvironment("TERM", "linux")
+                .withEnvironment("TERM", "xterm+256color")
```

Comment:
Unfortunately, switching to `xterm+256color` breaks coloring on macs. It looks like we need to provide different terminal types for different platforms.